sql/analyzer/resolve_columns.go

```diff
@@ -417,8 +417,16 @@ func getAvailableNamesByScope(n sql.Node, scope *Scope) availableNames {
 			return true
 		})
 		transform.Inspect(in.Source, func(n sql.Node) bool {
-			if resTbl, ok := n.(*plan.ResolvedTable); ok && !aliasedTables[resTbl] {
-				children = append(children, resTbl)
+			switch n := n.(type) {
+			case *plan.ResolvedTable:
+				if !aliasedTables[n] {
+					children = append(children, n)
+				}
+			case *plan.IndexedTableAccess:
+				if !aliasedTables[n] {
+					children = append(children, n.ResolvedTable)
+				}
+			default:
```

Comment:
Are there other implementations of the `sql.Table` that we should handle? Just wondering if we looked for `sql.Table` instead of the specific implementations if that would simplify this and cover any other edge cases that might show up.

Reply:
JSONTable, TableWrapper, and TableFunction are the three i'm worried about. I can add, but i had a hard time getting tests to match

Reply:
JSONTable and TableFunction as nested table source doesn't seem to trigger any errors. It is possible those are resolved directly, removing the need for the regular qualify and resolve paths to dig for hidden column sources.

Reply:
Sounds good; certainly not a blocker, just curious. Thanks for the extra details!